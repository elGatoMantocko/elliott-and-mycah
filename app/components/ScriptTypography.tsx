import { OverridableComponent, OverrideProps } from '@material-ui/core/OverridableComponent';
import { makeStyles } from '@material-ui/core/styles';
import Typography, { TypographyProps, TypographyTypeMap } from '@material-ui/core/Typography';
import * as React from 'react';

const useScriptStyles = makeStyles((theme) => ({
  root: {
    fontFamily: 'Lucian Schoenschrift CAT',
    fontSize: ({ variant }: TypographyProps) =>
      (variant &&
        variant !== 'inherit' &&
        variant !== 'srOnly' &&
        `calc(${theme.typography[variant].fontSize} * 1.2)`) ||
      26,
    position: 'relative',
    top: 4,
  },
}));

export const ScriptTypography: OverridableComponent<TypographyTypeMap> = <
  C extends React.ElementType = TypographyTypeMap['defaultComponent']
>(
  props: OverrideProps<TypographyTypeMap<unknown, C>, C>,
) => <Typography {...props} classes={useScriptStyles(props)} />;
