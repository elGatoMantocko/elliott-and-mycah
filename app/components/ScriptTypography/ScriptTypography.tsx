import { Typography, TypographyProps } from '@mui/material';
import { useTheme } from '@mui/material/styles';
import React from 'react';

export const ScriptTypography = (props: TypographyProps) => {
  const theme = useTheme();

  return (
    <Typography
      {...props}
      sx={{
        ...props.sx,
        fontFamily: '"Lucian Schoenschrift CAT", serif',
        fontSize:
          (props.variant &&
            props.variant !== 'inherit' &&
            `calc(${theme.typography[props.variant].fontSize} * 1.2)`) ||
          26,
        position: 'relative',
        top: 4,
      }}
    />
  );
};
