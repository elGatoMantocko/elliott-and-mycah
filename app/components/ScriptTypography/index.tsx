import { Typography, TypographyProps } from '@mui/material';
import { makeStyles } from '@mui/styles';
import React from 'react';

const useScriptStyles = makeStyles((theme) => ({
  root: {
    fontFamily: '"Lucian Schoenschrift CAT", serif',
    fontSize: ({ variant }: TypographyProps) =>
      (variant && variant !== 'inherit' && `calc(${theme.typography[variant].fontSize} * 1.2)`) ||
      26,
    position: 'relative',
    top: 4,
  },
}));

export const ScriptTypography = (props: TypographyProps) => (
  <Typography {...props} classes={useScriptStyles(props)} />
);
