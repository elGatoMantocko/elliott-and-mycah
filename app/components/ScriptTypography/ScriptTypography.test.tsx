import { render } from '@testing-library/react';
import React from 'react';

import { ScriptTypography } from './ScriptTypography';

describe('<ScriptTypography />', () => {
  it('should render', () => {
    const el = render(<ScriptTypography>Test copy</ScriptTypography>);
    expect(el.container.innerHTML).toMatchSnapshot();
  });
});
