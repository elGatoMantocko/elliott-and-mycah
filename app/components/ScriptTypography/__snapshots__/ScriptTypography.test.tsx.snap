// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1vt3dvp-MuiTypography-root"
  >
    Test copy
  </p>
</div>
`;

exports[`should render an h1 variant 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1whdp8y-MuiTypography-root"
  >
    Test copy
  </h1>
</div>
`;
